### PR TITLE
Beacon Security Object added

### PR DESCRIPTION
**Pull request by saifGeL**
Objects added for Query History and network information.

### CODE REVIEW

**Comment by garysaunders** [comment]
OK let's postpone to v1.0.1.

I will review once we have made the v1.0.0 release and this can be in the develop branch.

File: beacon.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.0
 servers: []
 info:
-  version: "0.4.1-SNAPSHOT"
+  version: "1.0.0"
```
Comment (mcupak) [inline beacon.yaml]: You should not change the version in this commit.

File: beacon.yaml [hunk 2]
```diff
@@ -231,6 +231,38 @@ paths:
...
+      parameters:
+        - in: path
+          name: dataset_id
+          schema:
+            type: integer
+          required: true
+          description: Numeric ID of the dataset to get
+
+
+
```
Comment (mcupak) [inline beacon.yaml]: Unnecessary whitespace.

File: beacon.yaml [hunk 3]
```diff
@@ -652,4 +740,4 @@ components:
             https://raw.githubusercontent.com/sdelatorrep/ADA-M/openapi_v2.0/adam.yaml#/definitions/AdamTerms
         metaConditions:
           $ref: >-
-            https://raw.githubusercontent.com/sdelatorrep/ADA-M/openapi_v2.0/adam.yaml#/definitions/AdamMetaConditions
+            https://raw.githubusercontent.com/sdelatorrep/ADA-M/openapi_v2.0/adam.yaml#/definitions/AdamMetaConditions
```
Comment (mcupak) [inline beacon.yaml]: Don't change this line (looks like just whitespace anyway?).

**Comment by mcupak** [comment]
Looks good overall, I requested some small changes. This is a very big update and should not be a part of 1.0.1, which would be a patch release. I'd say 1.1 or later, maybe even 2.0 because of how substantial this change is.

It also should not become a part of the standard before we have a demonstration of the fact that this actually works and can be reasonably implemented, e.g. it becomes a part of ELIXIR's implementation. I wonder if it should even be a separate service, since this kind of network control does not necessarily need to be done by a beacon itself, and has limited applicability (only ELIXIR right now, no other beacons seem to need it). WDYT?

File: beacon.yaml [hunk 4]
```diff
@@ -231,6 +231,38 @@ paths:
...
+      description: Get information about network membership from Beacon
+      operationId: GetNetwork
+      responses:
+        '200':
+          description: sucessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Beacon'
+  /mitigation_strategy{dataset_id}:
```
Comment (sdelatorrep) [inline beacon.yaml]: Is this a typo? It looks like the endpoint should be `/mitigation_strategy/{dataset_id}` but the second slash is missing.

File: beacon.yaml [hunk 5]
```diff
@@ -231,6 +231,38 @@ paths:
             schema:
               $ref: '#/components/schemas/BeaconAlleleRequest'
         required: true
+  /network:
+    get:
+      description: Get information about network membership from Beacon
+      operationId: GetNetwork
+      responses:
```
Comment (sdelatorrep) [inline beacon.yaml]: Should we also add error codes as requested in a similar situation #188?

File: beacon.yaml [hunk 6]
```diff
@@ -231,6 +231,38 @@ paths:
...
+          description: sucessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Beacon'
+  /mitigation_strategy{dataset_id}:
+    get:
+      description: Get a mitigation strategy applied to a particular dataset
+      operationId: GetMitigationStrategy
+      responses:
```
Comment (sdelatorrep) [inline beacon.yaml]: Same as above: should we also add error codes?

File: beacon.yaml [hunk 7]
```diff
@@ -491,6 +523,62 @@ components:
...
+    BeaconQueryLog:
+      type: object
+      required:
+        - id
+        - querySet
+      properties:
+        id:
+          type: string
+          description: Unique Identifier for query log
+        querySet:
```
Comment (sdelatorrep) [inline beacon.yaml]: I'm getting an error pasting this content into the Swagger Editor:
```
Errors

Semantic error at components.schemas.BeaconQueryLog.properties.querySet
Schemas with 'type: array', require a sibling 'items: ' field
```
I'm using https://swagger.io/tools/swagger-editor/

**Comment by sdelatorrep** [comment]
Hi! I've added some comments too.
It makes sense to me to wait until we have an implementation of this. 
By the way, any beacon can be attacked so they will need mitigation strategies anyway... won't they?

**Comment by jrambla** [comment]
Hi, I probably don't understand correctly the PR and branches choreography.
For me, adding this objects is a major think and all the ramifications must be considered. 
I'm having questions, e.g., on the naming of objects and the endpoints.
I would like to focus on this topic and review that we are suggesting a coherent solution for all different dimensions of it.

**Comment by mcupak** [comment]
Closing since this is out of date and @saifGeL has moved one. This will likely look very different and be a part of a major release.
